### PR TITLE
Added ?download=true option for downloading logs from the logviewer

### PR DESCRIPTION
**Pull request by solarkennedy**
This adds a new url option to set the content-disposition option for
logs. This header forces your browser to do a "save as" with the right
filename and such.

I haven't done any testing really.

cc @parshap

### CODE REVIEW

File: logviewer/api/handlers.go [hunk 1]
```diff
@@ -30,11 +30,16 @@ func LogHandler(w http.ResponseWriter, r *http.Request) {
 	if fileName == "" {
 		fileName = "stdout"
 	}
+	addDownloadHeader := false
+	downloadOption := r.URL.Query().Get("download")
```
Comment (sargun) [inline logviewer/api/handlers.go:34]: Couldn't this be:

```go
addDownloadHeader = r.URL.Query().Get("download") == "true"
```

File: logviewer/api/handlers.go [hunk 2]
```diff
@@ -57,6 +62,12 @@ func logHandler(w http.ResponseWriter, r *http.Request, containerID, fileName st
 		return
 	}
 	defer shouldClose(fout)
+
+	if addDownloadHeader {
+		basename := filepath.Base(fileName)
+		w.Header().Set("Content-Disposition", "attachment; filename="+basename)
```
Comment (sargun) [inline logviewer/api/handlers.go:68]: Is there any escaping necccessary? For example, if the filename has a ; in it?

**Review by sargun — APPROVED** [review]

File: logviewer/api/handlers.go (hunk 1 above)
Comment (solarkennedy) [reply logviewer/api/handlers.go:34]: Ah totally

File: logviewer/api/handlers.go (hunk 2 above)
Comment (solarkennedy) [reply logviewer/api/handlers.go:68]: Uhhh maybe?
https://stackoverflow.com/questions/93551/how-to-encode-the-filename-parameter-of-content-disposition-header-in-http

But modern docs:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

Sounds like you just need quotes, but whoops, I didn't end the quote here!

But yea, sounds like it is a bit of a mess here, but it doesn't sound like I need to urlencode the name.

But it does sound like I need to escape quotemarks

Comment (solarkennedy) [reply logviewer/api/handlers.go]: Fixed trailing quote and added a comment linking to the docs on this header

**Pull request merged**